lxc-to-lxd/main_migrate.go

```diff
@@ -0,0 +1,602 @@
+	for _, c := range conf {
+		parts := strings.SplitN(c, "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		key := strings.TrimSpace(parts[0])
+		val := strings.TrimSpace(parts[1])
+
+		switch key {
+		case "lxc.signal.halt", "lxc.haltsignal":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.signal.halt=%s\n", val)
+		case "lxc.signal.reboot", "lxc.rebootsignal":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.signal.reboot=%s\n", val)
+		case "lxc.signal.stop", "lxc.stopsignal":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.signal.stop=%s\n", val)
+		case "lxc.apparmor.allow_incomplete", "lxc.aa_allow_incomplete":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.apparmor.allow_incomplete=%s\n", val)
+		case "lxc.pty.max", "lxc.pts":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.pty.max=%s\n", val)
+		case "lxc.tty.max", "lxc.tty":
+			newConfig["raw.lxc"] += fmt.Sprintf("lxc.tty.max=%s\n", val)
+		}
+	}
+
+	// Setup the container creation request
+	req := api.ContainersPost{
+		Name: container.Name(),
+		Source: api.ContainerSource{
+			Type: "migration",
+			Mode: "push",
+		},
+	}
+	req.Config = newConfig
+	req.Devices = newDevices
+	req.Profiles = []string{"default"}
+
+	// Set the container architecture if set in LXC
+	fmt.Println("Processing container architecture configuration")
+	var arch string
+	value = getConfig(conf, "lxc.arch")
+	if value == nil {
+		fmt.Println("Couldn't find container architecture, assuming native")
+		arch = runtime.GOARCH
+	} else {
+		arch = value[0]
+	}
+
+	archId, err := osarch.ArchitectureId(arch)
+	if err != nil {
+		return err
+	}
+
+	req.Architecture, err = osarch.ArchitectureName(archId)
+	if err != nil {
+		return err
+	}
+
+	if storage != "" {
+		req.Devices["root"] = map[string]string{
+			"type": "disk",
+			"pool": storage,
+			"path": "/",
+		}
+	}
+
+	if debug {
+		out, _ := json.MarshalIndent(req, "", "  ")
+		fmt.Printf("LXD container config:\n%v\n", string(out))
+	}
+
+	// Create container
+	fmt.Println("Creating container")
+	if dryRun {
+		fmt.Println("Would create container now")
+	} else {
+		op, err := d.CreateContainer(req)
+		if err != nil {
+			return err
+		}
+
+		progress := utils.ProgressRenderer{Format: "Transferring container: %s"}
+		_, err = op.AddHandler(progress.UpdateOp)
+		if err != nil {
+			progress.Done("")
+			return err
+		}
+
+		rootfs, _ := getRootfs(conf)
+
+		err = transferRootfs(d, op, rootfs, rsyncArgs)
+		if err != nil {
+			return err
+		}
+
+		progress.Done(fmt.Sprintf("Container '%s' successfully created", container.Name()))
+	}
+
+	return nil
+}
+
+func convertNetworkConfig(container *lxc.Container, devices types.Devices) error {
+	networkDevice := func(network map[string]string) map[string]string {
+		if network == nil {
+			return nil
+		}
+
+		device := make(map[string]string, 0)
+		device["type"] = "nic"
+
+		// Get the device type
+		device["nictype"] = network["type"]
+
+		// Convert the configuration
+		for k, v := range network {
```

Comment:
Does this handle networks specified with indexes correctly? Like:
```
lxc.net.1.type =
lxc.net.4.name =
lxc.net.1.name =
```

Reply:
Good question. Might want to test this case.

Reply:
This particular case doesn't work due to the index being `4`. Change that to `2`, and it works.

Reply:
Well, in the new network config parser in liblxc you can have mixed indeces.

Reply:
You can have mixed indices, yes, but can you have gaps in network numbering?
A test failure I got early with this branch suggested that you couldn't. That is if you have `lxc.net.0`, `lxc.net.1` and `lxc.net.3` without any `lxc.net.2`, the `lxc.net.3` is getting ignored by the LXC config parser.